crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -216,6 +216,12 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
             let overrides = overrides
                 .build()
                 .with_code(proc_exit::sysexits::CONFIG_ERR)?;
+            if walk_policy.force_exclude() && path.is_file() {
+                match overrides.matched(path, false) {
```

Comment:
Something I need to double check is that some of these APIs expect you to call them as you recurse into a directory and return confusing results if you only call them on the final path.

Reply:
Hum... I think you're correct. If I have `foo/bar/baz.file` directory structure and `extend-exclude = ["bar/*.file"]` configuration, using `typos foo` will ignore the file properly but `typos foo/bar/baz.file --force-exclude` won't (using `extend-exclude = ["foo/bar/*.file"]` or `extend-exclude = ["baz.file"]` works, though). :confused: 

Not sure how to fix it. Maybe we should propose a feature to `ignore` crate so that it optionally checks root entry as well: [`walk.rs#L936-L938`](https://github.com/BurntSushi/ripgrep/blob/f608d4d9b3ab210b7e6964ca7d1d7dc9c077329e/crates/ignore/src/walk.rs#L936-L938)?
I don't know how likely is it to be accepted, knowing that `Walk` is supposed to be used with directory as argument (from the docs at least), and we're using it with file path as well.

Reply:
You likely need to do `path.ancestors()`.rev()`, calling `matched` on each

Reply:
Apologies, I think I just did something wrong during my test. The file is **not** excluded in any case, which is the expected behavior.